src/plan/generational/immix/global.rs

```diff
@@ -225,11 +236,13 @@ impl<VM: VMBinding> GenImmix<VM> {
             global_side_metadata_specs:
                 crate::plan::generational::new_generational_global_metadata_specs::<VM>(),
         };
-        let immix_space = ImmixSpace::new(plan_args.get_space_args(
-            "immix_mature",
-            true,
-            VMRequest::discontiguous(),
-        ));
+        let immix_space = ImmixSpace::new(
+            plan_args.get_space_args("immix_mature", true, VMRequest::discontiguous()),
+            ImmixSpaceArgs {
+                reset_log_bit_in_major_gc: false,
+                unlog_object_when_traced: false,
```

Comment:
shouldn't this be true?

Reply:
There is a slight difference.

In generational immix, we only need to unlog objects in GC when we copy an object to the mature/immix space (no matter from nursery or from the mature space). When we copy from nursery to mature, we do not invoke `ImmixSpace::trace_object`. So setting `unlog_object_when_traced` to `true` will not affect the log bit for those promoted object (as `ImmixSpace::trace_object` is not called on them). We use `CopySemantics::PromoteToMature` or `CopySemantics::Mature` to set the log bit in the `post_copy`.

In sticky immix, any object we trace in the immix space (no matter copied or not) is considered promoted, and we need to unlog it. For example, when we trace a nursery object, even if we do not copy it, it is 'promoted' to a mature object, and we need to unlog it. In this case, using `post_copy` won't capture all the cases here. So I added this option, and for sticky immix, we can just unlog every object when we trace it.

Reply:
I can add some comments to make it clear why we do not need it to be `true`.

Reply:
I added some comments to explain this.